pydantic/utils.py

```diff
@@ -465,19 +422,30 @@ def __init__(self, value: Any, items: Union['AbstractSetIntStr', 'MappingIntStrA
             self._items: Union['AbstractSetIntStr', 'MappingIntStrAny']
             self._type: Type[Union[set, dict]]  # type: ignore
 
-        # For further type checks speed-up
-        if isinstance(items, Mapping):
-            self._type = dict
-        elif isinstance(items, AbstractSet):
-            self._type = set
-        else:
-            raise TypeError(f'Unexpected type of exclude value {items.__class__}')
+        items = self._coerce_items(items)
+        self._type = dict
```

Comment:
Seems like it's redundant now?

Reply:
Sorry @MrMrRobat I should not have pushed this yet, this is a work in extreme progress ;).  And yes, I'm currently exploring the possibility of removing the necessity for this `self._type` attribute.

Reply:
Sure, it's fine 🙂